src/mem3_shards.erl

```diff
@@ -242,7 +242,49 @@ terminate(_Reason, #st{changes_pid=Pid}) ->
     exit(Pid, kill),
     ok.
 
-code_change(_OldVsn, #st{}=St, _Extra) ->
+code_change(1 = OldVsn, {st, MaxSize, CurSize, ChangesPid} = St, _Extra) ->
+    twig:log(notice, "~p code_change ~p ~p", [?MODULE, OldVsn, St]),
+    upgrade(MaxSize, CurSize, ChangesPid);
+
+code_change({down, 2} = OldVsn, St, _Extra) ->
+    {st, MaxSize, CurSize, ChangesPid, _, _} = St,
+    twig:log(notice, "~p code_change ~p ~p", [?MODULE, OldVsn, St]),
+    downgrade(MaxSize, CurSize, ChangesPid).
```

Comment:
let's keep a code change clause that matches on #st{} so we don't screw ourselves over on the next release.